libs/datamodel/core/src/transform/ast_to_dml/db/context.rs

```diff
@@ -27,6 +38,37 @@ impl<'a, 'ast> Context<'a, 'ast> {
         self.diagnostics.has_errors()
     }
 
+    /// We need special code for scalar field attribute validation, because the
+    /// attributes on a scalar field are the attributes on the scalar field
+    /// itself, plus the attributes on the type alias it may be using. That type
+    /// alias may be referencing another type alias that has attributes, and so
+    /// on transitively.
+    ///
+    /// Other than for this peculiarity, this method is identical to
+    /// `visit_attributes()`.
+    pub(super) fn visit_scalar_field_attributes(
+        &mut self,
+        model_id: ast::ModelId,
+        field_id: ast::FieldId,
+        scalar_field_type: ScalarFieldType,
+        f: impl FnOnce(&'_ mut Attributes<'ast>, &'_ mut Context<'ast>),
+    ) {
+        self.attributes.attributes.clear();
+        self.attributes
+            .attributes
+            .extend(self.db.ast[model_id][field_id].attributes.iter());
+
+        let mut scalar_field_type = scalar_field_type;
```

Comment:
I still think the parameter should be `mut` already. This is not super common pattern, and it's coming to the method as a value, so you don't mutate anything that might leak out from here.

Reply:
I agree, doing this.